VideoLocker/src/main/java/org/edx/mobile/util/FileUtil.java

```diff
@@ -48,8 +50,9 @@
      * @param fileOrDirectory The file/folder that needs to be deleted.
      * @param exceptions      Name of the file/folder that needs to be skipped while deletion.
      */
-    public static void deleteRecursive(File fileOrDirectory, List<String> exceptions) {
-        if (exceptions.contains(fileOrDirectory.getName())) return;
+    public static void deleteRecursive(@NonNull File fileOrDirectory,
+                                       @Nullable List<String> exceptions) {
+        if (exceptions != null && exceptions.contains(fileOrDirectory.getName())) return;
```

Comment:
I think it would be better to not have the `exceptions` parameter declared as `Nullable`, and force the caller to send an empty `List` instead, making for more concise code. Note that there is a shared constant immutable empty `List` available at `Collections.EMPTY_LIST` for this kind of scenario, so this wouldn't need to have the cost of temporarily created object instantiation either. If there are lost of callers that don't have exceptions, then there could be a convenience duplicate method without this parameter, which might call through to this method with the empty list itself.